benchmark/bls.cpp

```diff
@@ -0,0 +1,417 @@
+#include <benchmark.hpp>
+#include <eosio/chain/apply_context.hpp>
+#include <eosio/chain/webassembly/interface.hpp>
+#include <eosio/testing/tester.hpp>
+#include <test_contracts.hpp>
+#include <bls12-381/bls12-381.hpp>
+#include <random>
+
+using namespace eosio;
+using namespace eosio::chain;
+using namespace eosio::testing;
+using namespace bls12_381;
+
+// benchmark BLS host functions directly without relying on CDT wrappers.
+// to run a benchmarking session, in the build directory, type
+// benchmark/benchmark -f bls
+
+namespace eosio::chain {
+// placed in eosio::chain name space so that interface_in_benchmark
+// can access transaction_context's private member max_transaction_time_subjective.
+// interface_in_benchmark is a friend of transaction_context.
```

Comment:
Seems like you could make `eosio::test::interface_in_benchmark` a friend. Just need to forward declare.
Or `eosio::benchmark` either one.

```
namespace eosio::test {
   struct interface_in_benchmark; // for benchmark testing
}

friend struct test::interface_in_benchmark;
```

Reply:
Thanks. This make BLS benchmark code belongs to a single `eosio::benchmark` name space.